Technology-specific terminology (#45)
theri/draft-enghardt-panrg-path-properties: draft-irtf-panrg-path-properties.md

: An approximate calculation or assessment of the value of a property. An assessed property includes the reliability of the calculation or assessment. The notion of reliability depends on the property.
For example, a path property based on an approximate calculation may describe the expected median one-way latency of packets sent on a path within the next second, including the confidence level and interval. A non-numerical assessment may instead include the likelihood that the property holds.

## Terminology usage for specific technologies

> The terminology described in this document is intended to be general and applicable to existing and future path-aware technologies.

Reviewer: ```suggestion
The terminology defined in this document is intended to be general and applicable to existing and future path-aware technologies.
```
I think "define terminology" is more fitting than "describe terminology" in this context. (Also, I'm getting 42.5k Google hits for "define terminology" and just 5.5k for "describe terminology" :smile:)

Author: Agreed ;)

> This document attempts to unify terminology which is used inconsistently between different path-aware technologies.

Reviewer: I'm not sure about this sentence. Maybe we can delete it?

The introduction already gives the motivation for the common terminology:
"As terms related to paths have different meanings in different areas of networking, first, this document provides a common terminology to define paths, path elements, and path properties."

And I'm not sure if we really unify existing terminology - we did survey existing terminology, but "unify" sounds like a stronger implication that what the doc actually does, and I'm not sure if we even "attempt to unify".

If we want to add motivation to this section, how about adding the following to the first sentence above?
"[…] to provide a common language for the Path-Aware Networking Research Group."
Because I think that's actually the main motivation for having a generic common terminology that applies to different areas of networking.

Author: I like your suggestion better as well, let's change it.

Author: ```suggestion
```

> Using the terminology defined in this document, a path-aware technology can define its path elements on a specific level of abstraction.

Reviewer: ```suggestion
Using this terminology, a path-aware technology can define and consider specific path elements and path properties on a specific level of abstraction.
```
If the drop the second sentence, this sentence would follow after the first, so we can remove redundancy.
Also, added a "specific" to contrast with "generic".

Author: That sounds good.

> For example, a technology might define paths as sequence of IP routers as in source routing ({{?RFC1940}}), on different layers ({{?RFC1122}}), or as a collection of entities such as an AS or an ERO.

Reviewer: ```suggestion
For instance, a technology may define path elements as IP routers, e.g., in source routing ({{?RFC1940}}). Alternatively, it may consider path elements on a different layer of the Internet Architecture ({{?RFC1122}}) or independently, e.g., as a collection of entities, such as an AS or an ERO.
```

I think the original sentence listed different things that were not directly comparable, so I tried modifying it.
Also, as the previous sentence mentions "path elements" (and not "paths"), I think it's good to continue with path elements here.

Author: > Also, as the previous sentence mentions "path elements" (and not "paths"), I think it's good to continue with path elements here.

Agreed

For me using "independently" in this context is a bit confusing (I guess the meaning is independent from specific Internet Architecture layers). Maybe we can rephrase it?

```suggestion
For instance, a technology may define path elements as IP routers, e.g., in source routing ({{?RFC1940}}). Alternatively, it may consider path elements on a different layer of the Internet Architecture ({{?RFC1122}}) or as a collection of entities not tied to a specific layer, such as an AS or an ERO.
```
